x-pack/plugins/serverless_search/public/application/components/api_key/security_privileges_form.tsx

```diff
@@ -39,14 +49,54 @@ export const SecurityPrivilegesForm: React.FC<SecurityPrivilegesFormProps> = ({
           <p>{error}</p>
         </EuiText>
       )}
+      <EuiPanel hasShadow={false} color="subdued">
+        <EuiFlexGroup gutterSize="none" justifyContent="flexEnd" alignItems="baseline">
+          <EuiFlexItem grow={false}>
+            <EuiText size="xs">
+              <h4>
+                {i18n.translate('xpack.serverlessSearch.apiKey.privileges.boilerplate.label', {
+                  defaultMessage: 'Replace with boilerplate:',
+                })}
+              </h4>
+            </EuiText>
+          </EuiFlexItem>
+
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              data-test-subj="serverlessSearchSecurityPrivilegesFormReadOnlyButton"
+              onClick={() => setShowReadOnlyBoilerplate(true)}
```

Comment:
Basically just move the `READ_ONLY_BOILERPLATE` to this file or a constants file that can be imported here. Then updating these `onClick` to something like this:

```suggestion
              onClick={() => onChangeRoleDescriptors(READ_ONLY_BOILERPLATE)}
```

With that we shouldn't need many other changes to `create_api_key_flyout.tsx`, right?

Reply:
Yes. but with this approach, I think we will loose edited payload between button clicks. 

https://github.com/elastic/kibana/assets/55930906/f4e0b2da-d519-493b-909e-433755db072d

Reply:
That is the behavior I would expect, the label is "Replace with boilerplate" so I think its expected to replace the entire current contents of the editor with the boilerplate code.

@sphilipse you agree?

Reply:
Yeah I agree with @TattdCodeMonkey that not saving state between button clicks makes more sense.